modules/behaviors/DataBehavior.js

```diff
@@ -0,0 +1,608 @@
+     */
+    pull: function() {
+      var thisDataBehavior = this;
+      return this.__getIds()
+        .then(function(ids) {
+          return thisDataBehavior.privateCollection.trackAndPull(ids);
+        })
+        .then(this.__fetchSuccess, this.__fetchFailed);
+    },
+
+    /**
+     * Retrieves the ids for this data object and passes them off to the private collection's trackAndFetch() method.
+     * @method fetch
+     * @return {$.Deferred.Promise} a jquery deferred promise that resolves to the retrieved models.
+     */
+    fetch: function() {
+      var thisDataBehavior = this;
+      return this.__getIds()
+        .then(function(ids) {
+          return thisDataBehavior.privateCollection.trackAndFetch(ids);
+        })
+        .then(this.__fetchSuccess, this.__fetchFailed);
+    },
+
+    /**
+     * Retrieves the ids for this data object and passes them off to the private collection to track and then does a
+     * pull or a fetch based on the alwaysFetch property.  (pull is default if always fetch is true then it fetches instead).
+     * @method retrieve
+     * @return {$.Deferred.Promise} a jquery deferred promise that resolves to the retrieved models.
+     */
+    retrieve: function() {
+      if (this.alwaysFetch) {
+        return this.fetch();
+      } else {
+        return this.pull();
+      }
+    },
+
+    /**
+     * Listens for the change event on the ids property and, if triggered, re-fetches the data based on the new ids.
+     * @method listenToIdsPropertyChangeEvent
+     */
+    listenToIdsPropertyChangeEvent: function() {
+      if (!_.isUndefined(this.ids.property)) {
+        this.stopListeningToIdsPropertyChangeEvent();
+        var idsPropertyNameAndContext = this.__parseIdsPropertyNameAndContext();
+        var idsContext = idsPropertyNameAndContext.idsContext;
+        var canListenToEvents = idsContext && _.isFunction(idsContext.on);
+        if (canListenToEvents) {
+          this.__currentContextWithListener = idsContext;
+          this.__currentContextEventName = 'change:' + idsPropertyNameAndContext.idsPropertyName;
+          this.__dependsOnBehavior = idsPropertyNameAndContext.isBehavior;
+          if (this.__dependsOnBehavior) {
+            this.listenTo(this.__currentContextWithListener, 'fetched', this.retrieve);
+            this.listenTo(this.__currentContextWithListener.privateCollection, this.__currentContextEventName, this.retrieve);
+          } else {
+            this.listenTo(this.__currentContextWithListener, this.__currentContextEventName, this.retrieve);
+          }
+        }
+      }
+    },
+
+    /**
+     * Removes the listener added by listenToIdsPropertyChangeEvent().
+     * @method stopListeningToIdsPropertyChangeEvent
+     */
+    stopListeningToIdsPropertyChangeEvent: function() {
+      if (this.__currentContextWithListener) {
+        if (this.__dependsOnBehavior) {
+          this.stopListening(this.__currentContextWithListener, 'fetched', this.retrieve);
```

Comment:
yea data behaviors don't publish change events on themselves, but I think it might be useful?  If so then I can drop a lot of this stuff.  Instead of 'fetched' I could listen for 'change:context' and fire that along with 'fetched', then I could normalize this to a consistent block of code.